include/misc/util.h

```diff
@@ -576,4 +576,22 @@ static inline s64_t arithmetic_shift_right(s64_t value, u8_t shift)
 		     z_rep_5, z_rep_4, z_rep_3, z_rep_2, z_rep_1, z_rep_0) \
 	 (fixed_arg, x, ##__VA_ARGS__)}
 
+/*
+ * LOOP is used for a number of statements with incremented argument, can
+ * be used for defining number of serial ports or other devices
+ *
+ * Note: When taking an argument configuration parameters (CONFIG_NUM_OF_xxx)
+ * following construction might be used:
+ *
+ * UTIL_EVAL(UTIL_DEFER(LOOP) (statement, number))
```

Comment:
Seems like this "note" needs a little more explanation.  Recursive preprocessor expansion is deep voodoo.  I'd suggest a full, concrete example here and clearer text like for example:

    The n argument MUST be a literal decimal number in the range [1..4],
    so if you want it to be expanded from another preprocessor macro you
    must pre-expand the expression like this:

    UTIL_EVAL(UTIL_DEVER(LOOP)(MY_MACRO_TO_EXPAND, CONFIG_MY_NUMBER_OF_ITERATIONS)